devito/operator/operator.py

```diff
@@ -511,7 +522,7 @@ def _prepare_arguments(self, autotune=None, **kwargs):
                 # E.g., SubFunctions
                 continue
             for k, v in p._arg_values(**kwargs).items():
-                if k in args and args[k] != v:
+                if k in args and args[k] not in as_tuple(v) and k not in futures:
```

Comment:
why the `as_tuple` and wouldn't it be dangerous if `v` is an array (which we saw in some previous PR can happen in some very weird corner cases)

Reply:
now v can be a set of legal values such as `{14, 15, 16}` or a plain number such as `14`. I hadn't thought about the numpy case with is... crucial 😬 

so yes, gonna change this

as for why `v` can now be a set, it's part of the 1927 fix. Take a look at `ConditionalDimension._arg_defaults`

Reply:
actually... 

```
args[k] != v:
```

this, in master, would cause a whole-array comparison too??

Reply:
No it wouldn't it would complain that you need to use `.all()` in boolean context

Reply:
You en up here for example when you use a `SubFunction` with a weird name and it tries to compare coordinate values

Reply:
just a minor note that this was addressed on slack